### PR TITLE
Expose output commit

### PR DESCRIPTION
**Pull request by jdoliner**

### CODE REVIEW

File: doc/reference/pipeline_spec.md [hunk 1]
```diff
@@ -619,6 +617,24 @@ The root mount point is at `/pfs`, which contains:
...
+- `PACH_OUTPUT_COMMIT_ID` the id of the commit being outputted to.
+- For each input there will be an envirnment variable with the same name
+    defined to the path of the file for that input. For example if you are
+    accessing an input called foo from the path `/pfs/foo` which contains a file
+    called `bar` then the environment variable `foo` will have the value
+    `/pfs/foo/bar`. 
+
+In addition to these environment variables Kubernetes also injects others for
+Services that are running inside the cluster. These allow you to connect to
+outside those services, which can be powerful but also can be hard to reason
```
Comment (JoeyZwicker) [inline doc/reference/pipeline_spec.md]: Typo. Should be "connect to those outside services"

File: doc/reference/pipeline_spec.md [hunk 2]
```diff
@@ -619,6 +617,24 @@ The root mount point is at `/pfs`, which contains:
...
+about, as processing me be retried multiple times. Interaction with outside
```
Comment (JoeyZwicker) [inline doc/reference/pipeline_spec.md]: typo: "processing might be retried"?

File: doc/reference/pipeline_spec.md [hunk 3]
```diff
@@ -619,6 +617,24 @@ The root mount point is at `/pfs`, which contains:
...
+
+There are several environment variables that get injected into the user code
+before it runs. They are:
+
+- `PACH_JOB_ID` the id the currently run job.
+- `PACH_OUTPUT_COMMIT_ID` the id of the commit being outputted to.
+- For each input there will be an envirnment variable with the same name
+    defined to the path of the file for that input. For example if you are
+    accessing an input called foo from the path `/pfs/foo` which contains a file
+    called `bar` then the environment variable `foo` will have the value
```
Comment (JoeyZwicker) [inline doc/reference/pipeline_spec.md]: How does this work if there are multiple files in pfs/foo?

File: doc/reference/pipeline_spec.md (hunk 2 above)
Comment (JoeyZwicker) [reply doc/reference/pipeline_spec.md]: How does processing being retried make it hard to reason about? Perhaps explain with simple example?

File: doc/reference/pipeline_spec.md [hunk 4]
```diff
@@ -187,6 +187,10 @@ Lines need not end in newline characters.
 `transform.env` is a map from key to value of environment variables that will be
 injected into the container
 
+Note: there are environment variables that are automatically injected into the
```
Comment (JoeyZwicker) [inline doc/reference/pipeline_spec.md]: Did you decide not to offer the entire job-info? That's fine, just curious.

Comment (jdoliner) [reply doc/reference/pipeline_spec.md]: I'm still planning to do that but I wanted to just do this the quickest way to get it available for customers.

File: doc/reference/pipeline_spec.md [hunk 5]
```diff
@@ -619,6 +617,31 @@ The root mount point is at `/pfs`, which contains:
...
-### Output Formats
-
-PFS supports data to be delimited by line, JSON, or binary blobs.
+# Environment Variables
+
+There are several environment variables that get injected into the user code
+before it runs. They are:
+
+- `PACH_JOB_ID` the id the currently run job.
+- `PACH_OUTPUT_COMMIT_ID` the id of the commit being outputted to.
```
Comment (JoeyZwicker) [inline doc/reference/pipeline_spec.md]: Had a user ask if input commit could also be an Env Var. Of course, he know he can get it from the output commit + prov, but thought having it directly would be more convenient.

**Pull request merged**